sw/device/silicon_creator/lib/cert/cert_unittest.cc

```diff
@@ -200,14 +206,31 @@ TEST_F(CertTest, CertOutdated) {
   EXPECT_EQ(matches, kHardenedBoolFalse);
 }
 
+TEST_F(CertTest, CertOutdatedSerialNumberSizeMismatch) {
+  hardened_bool_t matches = kHardenedBoolFalse;
+  uint8_t old_length =
+      valid_dice_cert_bytes_[kCertX509Asn1SerialNumberLengthByteOffset];
+  valid_dice_cert_bytes_[kCertX509Asn1SerialNumberLengthByteOffset] = 19;
+  ExpectFlashInfoPageRead(&kFlashCtrlInfoPageUdsCertificate,
+                          /*offset=*/0, kCertX509Asn1FirstWordsWithSerialNumber,
+                          valid_dice_cert_words_, kErrorOk);
+  EXPECT_EQ(
+      cert_x509_asn1_check_serial_number(&kFlashCtrlInfoPageUdsCertificate,
+                                         expected_sn_bytes_, &matches),
+      kErrorOk);
+  EXPECT_EQ(matches, kHardenedBoolFalse);
+  valid_dice_cert_bytes_[kCertX509Asn1SerialNumberLengthByteOffset] =
+      old_length;
+}
+
```

Comment:
Should there be a test with a short serial number (19 bytes) that matches? Here the unit test don't really cover that as far as I can see.

Reply:
Ah, yes great idea thanks. I added one.